src/analog.rs

```diff
@@ -0,0 +1,50 @@
+//! Analog-digital conversion traits
+
+use nb;
+
+/// This trait denotes an object, e.g. a GPIO pin, that is configured for use as an ADC channel.
+/// Objects of this type are passed to ADC-mode-specific methods.
+pub trait AdcChannel {
+    /// Channel ID type
+    ///
+    /// e.g. u8 for ADC channels 1-n, or a (partid, pinid) tuple for something more complex.
+    type ID;
+
+    /// ID of the channel associated with this pin
+    const CHANNEL: Self::ID;
+}
+
+/// ADCs that sample on single channels per request, and do so at the time of the request.
+pub trait SingleMode<Word, Pin: AdcChannel> {
```

Comment:
Not sure how I feel about the name `Pin`. Maybe it `Channel`, or just `C` would be simpler.

And have you considered moving `Pin` to `read_channel`? That would make it more convenient to use the trait (one less argument to specify in parameter lists), but I see two potential problems:
- I'm not sure if that would allow an implementation to specify what `Pin` is in a `where` clause.
- It would be inconsistent with `Continuous`.

So yeah, not sure if this is a good idea, but I'm interested in knowing if you tried/considered it.

Reply:
I mostly kept it there for consistency with the other modes. Nothing's preventing moving `Pin` into the various functions. If it's cleaner/easier, I'm not opposed.

Reply:
I'm not opposed to keeping it where it is. Just wondering out loud.